2023-11-29-diving-into-jtag-part-2.md

```diff
@@ -0,0 +1,218 @@
+---
+
+title: Diving into JTAG protocol. Part 2 — Debugging.
+description:
+    "This article explores debugging microcontrollers using JTAG, with a special focus on the ARM Cortex-M architecture and the STM32F407VG microcontroller."
+author: Aliaksandr Kavalchuk
+tags: arm, mcu, debugging, debugger
+---
+
+<!-- excerpt start -->
+
+Part 2 of a JTAG deep-dive series gives an in-depth look at interacting with a microcontroller's memory and engaging with the processor core and debug registers.
+
+<!-- excerpt end -->
+
+{% include newsletter.html %}
+
+{% include toc.html %}
+
+![image](https://github.com/memfault/interrupt/assets/33341738/8993d42a-f524-48b5-97bc-83f6c6f89db3)
+
+As noted in my previous article [Diving into JTAG protocol. Part 1 — Overview](https://medium.com/@aliaksandr.kavalchuk/diving-into-jtag-protocol-part-1-overview-fbdc428d3a16), JTAG was initially developed for testing integrated circuits and printed circuit boards. However, its potential for debugging was realized over time, and now JTAG has become the standard protocol for microcontroller debugging. Many Firmware and Embedded engineers first encountered it in this particular context.
```

Comment:
Let's link back to our article https://interrupt.memfault.com/blog/diving-into-jtag-part1

Reply:
Excellent catch, can't believe I didn't do that! Thanks for the astute review